src/update.c

```diff
@@ -162,10 +162,10 @@ int add_included_manifests(struct manifest *mom, int current, struct list **subs
 
 	/* Pass the current version here, not the new, otherwise we will never
 	 * hit the Manifest delta path. */
-	if (add_subscriptions(subbed, subs, current, mom, false, 0) >= 0) {
-		ret = 0;
-	} else {
+	if (add_subscriptions(subbed, subs, current, mom, false, 0) & (add_sub_ERR | add_sub_BADNAME)) {
```

Comment:
This commit is a problem that isn't dependent from your Iterative Manifest implementation. We are going to have a problem in the current master branch too, right?
If so I would open another PR with this commit and have this merged now.

Reply:
yes, is unrelated... it may be a good idea to open a different PR just for this.